### PR TITLE
API v3 basic auth API doc

### PR DESCRIPTION
**Pull request by machisuji**
Proposal for basic auth for API v3. Contains only the documentation.

Open for discussion (/cc @opf/developers). Not intended for the 4.1 release, but would be nice for 4.2.

WP [#19294](https://community.openproject.org/work_packages/19294)

The argument about what to use is directly in the documentation currently.

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://machisuji-feature-api-v3-basic-auth-api-doc.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://machisuji-feature-api-v3-basic-auth-api-doc.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

File: doc/apiv3-documentation.apib [hunk 1]
```diff
@@ -40,11 +40,42 @@ in the [Collections Section](#collections).
...
+
+The simplest way to do basic auth would be to use a user's username and password naturally.
+However, OpenProject already has supported API keys in the past for the API v2, though not through basic auth.
+
+Using **username and password** directly would have some advantages:
+
+* It is intuitive for the user who then just has to provide those just as they would when logging into OpenProject.
+* No extra logic for token managment necessary.
+
+On the other hand using **API keys* has some advantages too, which is why we went for them:
```
Comment (NobodysNightmare) [inline doc/apiv3-documentation.apib]: broken formatting, two stars before `API keys`, one after.

File: doc/apiv3-documentation.apib [hunk 2]
```diff
@@ -40,11 +40,42 @@ in the [Collections Section](#collections).
...
+
+* If compromised while saved on a insecure client the user only has to regenerate the API key instead of changing their password, too.
```
Comment (NobodysNightmare) [inline doc/apiv3-documentation.apib]: on a**n** insecure client?

**Comment by NobodysNightmare** [comment]
@opf/developers Will merge this today if there are no further concerns.

**Comment by floriank** [comment]
This is fine by me as long we don't lose the ideas on other forms of authentication

**Comment by NobodysNightmare** [comment]
Haha, other what?
Merged!

**Pull request merged**